File: mvn-advent-of-code/src/test/java/advent/twenty_fifteen/Day06ProbablyAFireHazardTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day06ProbablyAFireHazardTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day06ProbablyAFireHazard("/twenty_fifteen/Day06-ProbablyAFireHazard-input.txt");

        assertEquals(569999, dp.getPart1Answer().intValue());
        assertEquals(17836115, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day06ProbablyAFireHazard("/twenty_fifteen/Day06-ProbablyAFireHazard-example#1.txt");

        assertEquals(998996, dp.getPart1Answer().intValue());
        assertEquals(1001996, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day06ProbablyAFireHazardTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day06ProbablyAFireHazardTest extends FlatSpec with Matchers {
  "2015 Day 6 - Input" should "calculate answers" in {
    val dp = new Day06ProbablyAFireHazard("/twenty_fifteen/Day06-ProbablyAFireHazard-input.txt")
    dp.part1Answer should be(569999)
    dp.part2Answer should be(17836115)
  }

  "2015 Day 6 - Example #1" should "calculate answers" in {
    val dp = new Day06ProbablyAFireHazard("/twenty_fifteen/Day06-ProbablyAFireHazard-example#1.txt")
    dp.part1Answer should be(998996)
    dp.part2Answer should be(1001996)
  }

}
